platform/darwin/src/NSValue+MGLStyleEnumAttributeAdditions.h

```diff
@@ -0,0 +1,150 @@
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLTextPitchAlignment:(MGLTextPitchAlignment)type;
+
+/**
+Creates a new value object containing the given `MGLTextRotationAlignment` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLTextRotationAlignment:(MGLTextRotationAlignment)type;
+
+/**
+Creates a new value object containing the given `MGLTextJustify` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLTextJustify:(MGLTextJustify)type;
+
+/**
+Creates a new value object containing the given `MGLTextAnchor` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLTextAnchor:(MGLTextAnchor)type;
+
+/**
+Creates a new value object containing the given `MGLTextTransform` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLTextTransform:(MGLTextTransform)type;
+
+/**
+Creates a new value object containing the given `MGLFillTranslateAnchor` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLFillTranslateAnchor:(MGLFillTranslateAnchor)type;
+
+/**
+Creates a new value object containing the given `MGLLineTranslateAnchor` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLLineTranslateAnchor:(MGLLineTranslateAnchor)type;
+
+/**
+Creates a new value object containing the given `MGLIconTranslateAnchor` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLIconTranslateAnchor:(MGLIconTranslateAnchor)type;
+
+/**
+Creates a new value object containing the given `MGLTextTranslateAnchor` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLTextTranslateAnchor:(MGLTextTranslateAnchor)type;
+
+/**
+Creates a new value object containing the given `MGLCircleTranslateAnchor` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLCircleTranslateAnchor:(MGLCircleTranslateAnchor)type;
+
+/**
+Creates a new value object containing the given `MGLCirclePitchScale` structure.
+
+@param type The value for the new object.
+@return A new value object that contains the style enumeration type.
+*/
++ (instancetype)valueWithMGLCirclePitchScale:(MGLCirclePitchScale)type;
+
+
```

Comment:
Each factory method should come with a readonly property that converts back into the raw type ([example](https://github.com/mapbox/mapbox-gl-native/blob/307f8e4f41a440fdfcd9c7bb1833110ab9278ccf/platform/darwin/src/NSValue%2BMGLAdditions.h#L27)). This way you (and the developer) won’t ever have to use `-getValue:`.